### ISSUE TITLE
Accidental Double-Click in Proof Tree During Proof crashes GUI (proof tree)

### ISSUE THREAD

**Issue by wadoon** [KeYProject/key#402]
This issue was created at [git.key-project.org](https://git.key-project.org/key/key/-/issues/402) where the discussions are preserved.
    
----
    
* Mantis: [MT-1556](http://i12www.ira.uka.de/~klebanov/mantis/view.php?id=1556)
* Submitted on: 2015-07-28 by (at)mkirsten
* Updated: 2015-07-28

### Description

> When you accidentally double-click on the text in a goal on the proof tree (which tries to edit the goal tag/name) during the proof, you cannot access the proof tree any more and the attached exception is thrown.

### Steps to reproduce

```
See above.
```

### Additional Information

```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at javax.swing.plaf.basic.BasicTreeUI.stopEditing(BasicTreeUI.java:597)
	at javax.swing.plaf.basic.BasicTreeUI$Handler.mousePressed(BasicTreeUI.java:3536)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:280)
	at java.awt.Component.processMouseEvent(Component.java:6522)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6290)
	at java.awt.Container.processEvent(Container.java:2234)
	at java.awt.Component.dispatchEventImpl(Component.java:4881)
	at java.awt.Container.dispatchEventImpl(Container.java:2292)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4898)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4530)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4462)
	at java.awt.Container.dispatchEventImpl(Container.java:2278)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

## History

* (at)mkirsten -- (`NEW_BUG`) _2015-07-28_

## Attributes

* Category: GUI
* Status: NEW
* Severity: BLOCK
* OS: 
* Target Version: 
* Resolution: OPEN
* Priority: NORMAL
* Reproducibility: ALWAYS
* Platform: 
* Commit: 33508c8c3f337e4d9d8c7290dc84ccbbdd96414c
* Build: 
* Tags []
* Labels:  ~GUI  ~Bug  ~NORMAL  

---

[View in Mantis](http://i12www.ira.uka.de/~klebanov/mantis/view.php?id=1556)
    
----
    
Information: 
* created_at: 2017-05-29T02:36:42.103Z
* updated_at: 2017-05-29T02:36:42.103Z
* closed_at: None  (closed_by: )
* milestone: 
* user_notes_count: 0

**Comment by FliegendeWurst** [comment]
Fix in #3168

**Comment by jwiesler** [comment]
Open since 2015. Jeez.
